Library/Homebrew/formula.rb

```diff
@@ -261,6 +261,15 @@ def installed?
     (dir = installed_prefix).directory? && dir.children.length > 0
   end
 
+  # If this {Formula} has been modified by the user.  Relies on git.
+  def adulterated?
+    repo = core_formula? ? HOMEBREW_LIBRARY : path_of_repository
+    repo.cd do
+      return true unless `git status --porcelain #{path}`.empty?
```

Comment:
Hmm, it might be best to just merge #41397; I added this because I had it ready and lack the experience with Homebrew to know which was better.